### PR TITLE
dnsmasq: update default configuration

### PR DESCRIPTION
**Pull request by gregdel**
This now works ! The trick is to disable the authoritative mode from the dnsmasq configuration.
It works if we keep the NAK patch from @sduponch

### CODE REVIEW

File: overthebox/files/etc/uci-defaults/otb-dnsmasq.defaults [hunk 1]
```diff
@@ -33,7 +10,20 @@ if ! uci show dhcp | grep 'overthebox' 1>/dev/null 2>/dev/null; then
 	uci set dhcp.@cname[-1].target='lan'
 fi
 
-uci commit dhcp
+uci -q batch <<-EOF
+set dhcp.@dnsmasq[0].cachesize='8192'
+set dhcp.@dnsmasq[0].filter_aaaa='1'
+set dhcp.@dnsmasq[0].all_servers='1'
+set dhcp.@dnsmasq[0].authoritative='0'
```
Comment (gregdel) [inline overthebox/files/etc/uci-defaults/otb-dnsmasq.defaults]: This is the only line changed from my previous PR

**Comment by gregdel** [comment]
I still need to patch the dnsmasq config upstream tho, see this PR:
https://github.com/lede-project/source/pull/1174

File: overthebox/files/etc/uci-defaults/otb-firewall.defaults [hunk 2]
```diff
@@ -18,6 +18,16 @@ _setup() {
 config_load firewall
 config_foreach _setup zone
 
+uci -q batch <<-EOF
+add firewall rule
```
Comment (angt) [inline overthebox/files/etc/uci-defaults/otb-firewall.defaults]: This rule will be added at each upgrade. You should name it:
```
set firewall.allow_dhcp_request=rule
set firewall.allow_dhcp_request.src=wan
...
```

**Comment by sduponch** [comment]
Ok, we are with the same workaround we use on the actual OverTheBox dnsmasq configuration.

My PR that use an iptables workaround, does not need any extra uci config, allow autoritative mode and does not require dhcp to be forced.

**Comment by sduponch** [comment]
I think you will have a router surprise with lan, if1, if2 on the same network range.

**Comment by gregdel** [comment]
Are you sure ? Only the LAN is dynamic

**Comment by sduponch** [comment]
On previous version, i have to play with "tag-if" and "dhcp-mac" to help dnsmasq to select the good tag, as networkid does not help.

**Comment by gregdel** [comment]
I removed the **WIP**, it has been tested on windows, linux and macOS.
I think this is can be **rebased** ;)

**Comment by gregdel** [comment]
BTW, this fixes #438

**Comment by sduponch** [comment]
root@OpenWrt:/usr/lib/qos# ping overthebox.ovh
PING overthebox.ovh (192.168.0.20): 56 data bytes
64 bytes from 192.168.0.20: seq=0 ttl=64 time=0.103 ms
64 bytes from 192.168.0.20: seq=1 ttl=64 time=0.116 ms

**Comment by gregdel** [comment]
What's your point ?

**Comment by gregdel** [comment]
I removed the cname since we can use the `interface_name` option on the dhcp-range of the lan
I also removed `if0` from the interfaces to listen on

**Pull request merged**